system/security/securesocket/securesocket.hpp

```diff
@@ -30,6 +30,7 @@
 
 #include "jsocket.hpp"
 #include "jptree.hpp"
+#include "jsmartsock.ipp"
```

Comment:
You should be able to move this include out of the hpp and into securesocket.cpp afaics